hello/hello_signal.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+from typing import List
+
+from temporalio import workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+
+@workflow.defn
+class GreetingWorkflow:
+    def __init__(self) -> None:
+        self._pending_greetings: asyncio.Queue[str] = asyncio.Queue()
+        self._exit = False
+
+    @workflow.run
+    async def run(self) -> List[str]:
+        # Continually handle from queue or wait for exit to be received
+        greetings: List[str] = []
+        while not self._exit:
+            # Wait for queue item or exit
+            await workflow.wait_condition(
+                lambda: not self._pending_greetings.empty() or self._exit
+            )
+
+            # Drain and process queue
+            while not self._pending_greetings.empty():
+                greetings.append(f"Hello, {self._pending_greetings.get_nowait()}")
+
+        return greetings
+
+    @workflow.signal
+    async def submit_greeting(self, name: str) -> None:
```

Comment:
Doesn't need to be async right?
I'm considering disallowing async signal handler in TS altogether because users don't understand that they need to wait for all handlers before completing their workflow and end up losing progress.

It's much easier to reason about workflow code if all "io" is done in the main workflow thread.

Reply:
It does need to be async because I use `await` for putting on the queue. I could use `put_nowait` since the queue does not have a max size, but this is clear too.

> It's much easier to reason about workflow code if all "io" is done in the main workflow thread.

A caller choice IMO.

Reply:
Yes, right now it's the caller's choice but it's also way too easy to get wrong.